### ISSUE TITLE
init command duplicates node_modules gitignore entry

### ISSUE THREAD

**Issue by namoscato** [tinacms/tinacms#3490]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

[`checkGitignoreForNodeModules`](https://github.com/tinacms/tinacms/blob/%40tinacms/cli%401.0.5/packages/@tinacms/cli/src/cmds/init/index.ts#L279-L288) is not properly accounting for scenarios in which the `node_modules` gitignore entry is prefixed and/or suffixed with a forward slash.

### Reproduction

https://github.com/namoscato/todaypgh

### Steps to reproduce

1. given an existing repository with `.gitignore`, i.e.

    ```
    /node_modules/
    ```

1. run `npx @tinacms/cli@latest init`
2. see that `node_modules` entry is incorrectly added

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 159.88 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
    Yarn: 1.22.19 - ~/dev/git/todaypgh/node_modules/.bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.0/bin/npm
  Browsers:
    Chrome: 108.0.5359.124
    Chrome Canary: 111.0.5508.0
    Firefox: 104.0.2
    Safari: 16.2
  npmPackages:
    @tinacms/cli: ^1.0.5 => 1.0.5 
    tinacms: ^1.1.2 => 1.1.2
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by jamespohalloran** [comment]
To confirm I understand, is the issue that it adds a redundant `node_modules`, when you already have a `/node_modules` or `/node_modules/` in your .gitignore?

**Comment by namoscato** [comment]
Yeah, correct. Relatively minor issue but figured I would make a note.
